cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -95,6 +96,9 @@ public void download(String url, Path target) {
     this.context.info("Trying to download {} from {}", target.getFileName(), url);
     mkdirs(target.getParent());
     try {
+      if (context.isOffline()) {
+        throw new CliOfflineException("Not able to download because we are offline");
+      }
```

Comment:
The `CliOfflineException` already has a standard message in its default constructor.
If we want to have a custom message for downloads, I would suggest that you provide this also in `CliOfflineException` as an additional String constructor taking the `url` as argument and giving some context information so we can also see which tool this was and what version and edition since all is somehow in the URL. Ideally you make the constructor private and expose it as a static "factory" method with a speaking name `public static CliOfflineException ofDownload(String url) { return new CliOfflineException(url); }` with
```java
private CliOfflineException(String url) {
  super("You are offline and cannot download from URL " + url);
}
```